From 4828757e862d9f69cd70827aeef7ec3163f51b6a Mon Sep 17 00:00:00 2001
From: Colin Patrick Mccabe <cmccabe@cloudera.com>
Date: Tue, 31 May 2016 17:19:25 -0700
Subject: [PATCH 1826/2748] HDFS-10415.
 TestDistributedFileSystem#MyDistributedFileSystem
 attempts to set up statistics before initialize()
 is called (Mingliang Liu via cmccabe)

(cherry picked from commit 14907d9b8cf5ad7a8241aa66db79cc8e79767cbe)
(cherry picked from commit 0d6a65a40abca13230e93b871f3a8c345bac9377)

Change-Id: I3ec7f75929c9aac1ae4571003029bc6dbf5fd3c2
---
 .../hadoop/hdfs/TestDistributedFileSystem.java     |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
index e906e08..cd3126b 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
@@ -217,13 +217,19 @@ public void testDFSCloseOrdering() throws Exception {
   
   private static class MyDistributedFileSystem extends DistributedFileSystem {
     MyDistributedFileSystem() {
-      statistics = new FileSystem.Statistics("myhdfs"); // can't mock finals
       dfs = mock(DFSClient.class);
     }
     @Override
     public boolean exists(Path p) {
       return true; // trick out deleteOnExit
     }
+    // Symlink resolution doesn't work with a mock, since it doesn't
+    // have a valid Configuration to resolve paths to the right FileSystem.
+    // Just call the DFSClient directly to register the delete
+    @Override
+    public boolean delete(Path f, final boolean recursive) throws IOException {
+      return dfs.delete(f.toUri().getPath(), recursive);
+    }
   }
 
   @Test
-- 
1.7.9.5

